lib/simulator-management.js

```diff
@@ -6,16 +6,16 @@ import _ from 'lodash';
 import log from './logger';
 
 // returns sim for desired caps
-async function createSim (caps, sessionId) {
-  let name = `appiumTest-${sessionId}`;
+async function createSim (caps) {
+  let name = `appiumTest-${caps.deviceName}`;
   let udid = await createDevice(name, caps.deviceName, caps.platformVersion);
   return await getSimulator(udid);
 }
 
 async function getExistingSim (opts) {
   let devices = await getDevices(opts.platformVersion);
   for (let device of _.values(devices)) {
-    if (device.name === opts.deviceName) {
+    if ((device.name === opts.deviceName) || (device.name === `appiumTest-${opts.deviceName}`)) {
```

Comment:
This seems like it would change the behavior of the system. If, for instance, I had `deviceName` of `iPhone 7`, but on my system there are two simulators, `iPhone 7` (the one I want) and `appiumTest-iPhone 7`, which one I get depends on the order simctl returns them in the list, and the order that `_.values` returns them when iterating (something that is _not_ deterministic).

I would be fine with saving the `appiumTest-` device as an alternative, if the actual device I am requesting is not found, but this current way is broken.

Reply:
I see > the order

Is the below expected?

1. Check `device.name === opts.deviceName` and if device pool has the name, `getExistingSim` returns the device.
2. Check `appiumTest` prefix since nothing matches with `device.name === opts.deviceName`.

Test code is: https://github.com/appium/appium-xcuitest-driver/pull/674/files#diff-7dd0b2f2d113b4a83de18822c3ed06b4R57
(The test fails with the previous implementation)